src/axom/quest/DistributedClosestPoint.cpp

```diff
@@ -205,99 +204,110 @@ void DistributedClosestPoint::setObjectMesh(const conduit::Node& meshNode,
-    break;
-  case 3:
-    m_dcp_3->importObjectPoints(mdMeshNode, topologyName);
-    break;
-  }
+  m_impl->importObjectPoints(mdMeshNode, topologyName);
 
   return;
 }
 
 bool DistributedClosestPoint::generateBVHTree()
 {
-  SLIC_ASSERT_MSG(m_objectMeshCreated,
+  SLIC_ASSERT_MSG(m_impl,
                   "Must call 'setObjectMesh' before calling generateBVHTree");
 
-  bool success = false;
-
-  // dispatch to implementation class over dimension
-  switch(m_dimension)
-  {
-  case 2:
-    success = m_dcp_2->generateBVHTree();
-    break;
-  case 3:
-    success = m_dcp_3->generateBVHTree();
-    break;
-  }
-
+  bool success = m_impl->generateBVHTree();
   return success;
 }
 
 void DistributedClosestPoint::computeClosestPoints(conduit::Node& query_node,
                                                    const std::string& topology)
 {
-  SLIC_ASSERT_MSG(m_objectMeshCreated,
+  SLIC_ASSERT_MSG(m_impl,
                   "Must call 'setObjectMesh' before calling generateBVHTree");
 
   SLIC_ASSERT(this->isValidBlueprint(query_node));
 
-  // dispatch to implementation class over dimension
-  switch(m_dimension)
-  {
-  case 2:
-    m_dcp_2->setSquaredDistanceThreshold(m_sqDistanceThreshold);
-    m_dcp_2->setMpiCommunicator(m_mpiComm);
-    m_dcp_2->setOutputSwitches(m_outputRank,
-                               m_outputIndex,
-                               m_outputDistance,
-                               m_outputCoords,
-                               m_outputDomainIndex);
-    m_dcp_2->computeClosestPoints(query_node, topology);
-    break;
-  case 3:
-    m_dcp_3->setSquaredDistanceThreshold(m_sqDistanceThreshold);
-    m_dcp_3->setMpiCommunicator(m_mpiComm);
-    m_dcp_3->setOutputSwitches(m_outputRank,
-                               m_outputIndex,
-                               m_outputDistance,
-                               m_outputCoords,
-                               m_outputDomainIndex);
-    m_dcp_3->computeClosestPoints(query_node, topology);
-    break;
-  }
+  m_impl->setSquaredDistanceThreshold(m_sqDistanceThreshold);
+  m_impl->setMpiCommunicator(m_mpiComm);
+  m_impl->setOutputSwitches(m_outputRank,
+                            m_outputIndex,
+                            m_outputDistance,
+                            m_outputCoords,
+                            m_outputDomainIndex);
+  m_impl->computeClosestPoints(query_node, topology);
 }
 
 void DistributedClosestPoint::allocateQueryInstance()
 {
-  SLIC_ASSERT_MSG(m_objectMeshCreated == false, "Object mesh already created");
-
-  switch(m_dimension)
+  switch(m_runtimePolicy)
   {
-  case 2:
-    m_dcp_2 =
-      std::make_unique<internal::DistributedClosestPointImpl<2>>(m_runtimePolicy,
-                                                                 m_allocatorID,
-                                                                 m_isVerbose);
-    m_objectMeshCreated = true;
+  case RuntimePolicy::seq:
```

Comment:
Would it be better to set the execution policy type based on `m_runtimePolicy` and then use that in a single instance of the call to the DistributedClosestPointExec constructor? One way would be to have a method templated on dimension and exec policy that sets `m_impl` and call that in each case of the switch statement. That would simplify this a lot.